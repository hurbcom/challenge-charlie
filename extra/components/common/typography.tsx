import React from "react";
import {Typography} from "@material-ui/core";
import type { TypographyProps } from "@material-ui/core";

export function TypographyCustom(props: TypographyProps<"div">){
    return(
        <Typography style={props.style ?? {cursor:'default'}}
                    {...props}>
            {props.children}
        </Typography>
    )
}