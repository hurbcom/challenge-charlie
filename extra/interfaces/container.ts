import React from "react";

export interface ContainerInterface {
    children: React.ReactNode
}
