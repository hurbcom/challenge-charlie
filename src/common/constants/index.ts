export * from './temperature'
export * from './day'
export * from './wind'
export * from './endpoints'
