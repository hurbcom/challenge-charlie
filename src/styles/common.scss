.row {
    display: flex;
    flex-direction: row;
    align-items: center;
    align-content: center;
    flex: 1;
    width: 100%;
}

.column {
    display: flex;
    flex-direction: column;
    flex-basis: 100%;
    flex: 1;
    width: 100%;
}

.padding-10 {
    box-sizing: border-box;
    padding: 10px;
}

.left {
    justify-content: end;
}

.center {
    align-items: center;
    align-content: center;
    justify-content: center;
}

.margin-right-10 {
    margin-right: 10px;
}

.lower-case {
    text-transform: lowercase;
}
