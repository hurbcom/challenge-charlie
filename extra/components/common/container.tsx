import React from 'react';
import {Box} from "@material-ui/core";

//Interface
import {ContainerInterface} from "../../interfaces/container";

export function Container({children}: ContainerInterface){
    return(
        <Box>
            {children}
        </Box>
    )
}
