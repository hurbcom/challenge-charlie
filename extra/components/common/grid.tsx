import React from "react";
import {Grid} from "@material-ui/core";

//Hooks
import {useWindowSize} from "../../hooks/useWindowsSize";

//Interfaces
import {GridInterface, GridInterfaceNumber} from "../../interfaces/grid";

export function GridNumber (props:GridInterfaceNumber){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={props.number}
            xl={props.number}
            sm={props.number}
            md={props.number}
            lg={props.number}>
            {props.children}
        </Grid>
    )
}



export function Grid1 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid1}
            xl={size.grid1}
            sm={size.grid1}
            md={size.grid1}
            lg={size.grid1}>
            {props.children}
        </Grid>
    )
}

export function Grid2 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid2}
            xl={size.grid2}
            sm={size.grid2}
            md={size.grid2}
            lg={size.grid2}>
            {props.children}
        </Grid>
    )
}

export function Grid3 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid3}
            xl={size.grid3}
            sm={size.grid3}
            md={size.grid3}
            lg={size.grid3}>
            {props.children}
        </Grid>
    )
}

export function Grid4 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid4}
            xl={size.grid4}
            sm={size.grid4}
            md={size.grid4}
            lg={size.grid4}>
            {props.children}
        </Grid>
    )
}

export function Grid5 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid5}
            xl={size.grid5}
            sm={size.grid5}
            md={size.grid5}
            lg={size.grid5}>
            {props.children}
        </Grid>
    )
}

export function Grid6 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid6}
            xl={size.grid6}
            sm={size.grid6}
            md={size.grid6}
            lg={size.grid6}>
            {props.children}
        </Grid>
    )
}

export function Grid7 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid7}
            xl={size.grid7}
            sm={size.grid7}
            md={size.grid7}
            lg={size.grid7}>
            {props.children}
        </Grid>
    )
}

export function Grid8 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid8}
            xl={size.grid8}
            sm={size.grid8}
            md={size.grid8}
            lg={size.grid8}>
            {props.children}
        </Grid>
    )
}

export function Grid9 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid9}
            xl={size.grid9}
            sm={size.grid9}
            md={size.grid9}
            lg={size.grid9}>
            {props.children}
        </Grid>
    )
}

export function Grid10 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid10}
            xl={size.grid10}
            sm={size.grid10}
            md={size.grid10}
            lg={size.grid10}>
            {props.children}
        </Grid>
    )
}

export function Grid11 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid11}
            xl={size.grid11}
            sm={size.grid11}
            md={size.grid11}
            lg={size.grid11}>
            {props.children}
        </Grid>
    )
}

export function Grid12 (props:GridInterface){
    const size = useWindowSize()
    return(
        <Grid
            {...props}
            container
            item
            justifyContent={props.justifyContent ?? size.webFlexStart}
            xs={size.grid12}
            xl={size.grid12}
            sm={size.grid12}
            md={size.grid12}
            lg={size.grid12}>
            {props.children}
        </Grid>
    )
}