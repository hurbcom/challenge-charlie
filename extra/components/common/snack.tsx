import React, {useEffect} from 'react';
import {toast, ToastContainer} from "react-toastify";
import 'react-toastify/dist/ReactToastify.css';

//Redux
import {useAppSelector} from "../../redux/hooks";
import {selectSnack} from "../../redux/features/snack";

//Interfaces
import {ToastInterface} from "../../interfaces/snack";

const ToastMessage = ({ type, message }: ToastInterface) => {
    switch (type) {
        case 'success':
            toast.success(message);
            break;
        case 'warn':
            toast.warn(message);
            break;
        case 'error':
            toast.error(message);
            break;
        default:
            toast.info(message);
    }
}

export default function CustomizedSnackbars (){
    const state = useAppSelector(selectSnack);

    const notify = React.useCallback((type, message) => {
        ToastMessage({ type, message });
    }, []);

    useEffect(()=>{
        if(state.open){
            notify(state.type, state.message)
        }
    },[state])

    return (
        <ToastContainer
            position="top-right"
            autoClose={10000}
            hideProgressBar={false}
            newestOnTop
            closeOnClick
            rtl={false}
            theme={'colored'}
            pauseOnFocusLoss
            draggable
            pauseOnHover
        />
    );
}


