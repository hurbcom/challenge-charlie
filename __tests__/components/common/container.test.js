import React from 'react';
import renderer from 'react-test-renderer';

//Components
import {Container} from "../../../extra/components/common/container";

test('Container Component', () => {
    const component = renderer.create(
        <Container>
            <div id={'test'}/>
        </Container>
    );
    let tree = component.toJSON();
    //Snapshot
    expect(tree).toMatchSnapshot();
    //Value props children
    expect(tree.children[0].props.id).toEqual('test');

});