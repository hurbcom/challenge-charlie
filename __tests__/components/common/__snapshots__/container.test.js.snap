// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <div
    id="test"
  />
</div>
`;
